.github/workflows/e2e.yaml

```diff
@@ -116,21 +116,154 @@ jobs:
+      - name: install nginx
+        run: |
+          make init-nginx
+      - name: install workflows
+        run: |
+          make init-argo-workflows
+      - name: adding license env var
+        run: |
+          echo "GITLAB_LICENSE=${{ secrets.GITLABLICENSE }}" >> $GITHUB_ENV
+      - name: install gitlab
+        run: |
+          echo "GITLAB_LICENSE is $GITLAB_LICENSE"  # Debugging step
+          make init-gitlab
+      - name: populate gitlab
+        run: |
+          output=$(sh ./scripts/setup-gitlab.sh | tail -n1)
+          echo "GITLAB_TOKEN=$(echo $output | cut -d' ' -f1)" >> $GITHUB_ENV
+      - name: Build Docker Image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: true
+          tags: localhost:5001/piper:latest
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: Check tunnel existence
+        run: |
+          echo "NGROK_URL=$(cat ~/ngrok.log | grep -o 'url=https://.*' | cut -d '=' -f 2)" >> $GITHUB_ENV
+          cat ~/ngrok.log | grep -o 'url=https://.*' | cut -d '=' -f 2
+      - name: init piper
+        run: |
+          helm upgrade --install piper ./helm-chart \
+          -f ./examples/template.values.dev.yaml \
+          --set piper.gitProvider.name="gitlab" \
+          --set piper.gitProvider.token="${{ env.GITLAB_TOKEN }}" \
+          --set piper.gitProvider.url="http://gitlab-webservice-default.gitlab:8080" \
+          --set piper.gitProvider.webhook.url="${{ env.NGROK_URL }}/piper/webhook" \
+          --set piper.gitProvider.webhook.repoList={piper-e2e-test} \
+          --set piper.gitProvider.organization.name="pied-pipers" \
+          --set image.repository=localhost:5001 \
+          --set piper.argoWorkflows.server.address="${{ env.NGROK_URL }}/argo" \
+          --set-string env\[0\].name=GIT_WEBHOOK_AUTO_CLEANUP,env\[0\].value="true" && \
+          sleep 20 && kubectl logs deployment/piper 
+          kubectl wait \
+          --for=condition=ready pod \
+          --selector=app=piper \
+          --timeout=60s
+      - uses: actions/checkout@v3
+        with:
+          repository: "quickube/piper-e2e-test"
+          path: piper-e2e-test
+          ref: "main"
+      - name: inject some changes to piper-e2e-test repo
+        run: |
+          mkdir ./gitlab
+          cd ./gitlab
+          git clone http://oauth2:${{ env.GITLAB_TOKEN }}@localhost:8080/pied-pipers/piper-e2e-test.git
+          cp -r ../piper-e2e-test/.workflows ./piper-e2e-test/
+          cd ./piper-e2e-test
+          git config user.name 'piper-user'
+          git config user.email 'piper@example.com'
+          git add -A
+          git commit -m "add stuff"
+          git push
+          git checkout -b ${{ github.ref_name }}-test      
+          rm ./.workflows/triggers.yaml
+          cat <<EOF > ./.workflows/triggers.yaml
+          - events:
+              - merge_request
+              - merge_request.open
```

Comment:
maybe we need here merge_request.updated too

Reply:
is it needed? as the commit that triggers the pipe for the test is merge_request.open,